forms-web-app/src/validators/appellant-submission/site-location.js

```diff
@@ -1,5 +1,20 @@
 const { body } = require('express-validator');
 
+function validatePostcode(postcode) {
+  const firstChar = postcode.charAt(0);
+
+  if (!Number.isNaN(parseInt(firstChar, 10))) {
```

Comment:
I'm happy with the implementation of the stepdefs above,
I'm not sure about these as validation steps though?

We only put 2 example 'bad postcodes' in as an example of things we thought should fail, it's not supposed to be an exhaustive list (as things stand.. maybe it should be?)

My assumption at the time was: there will be a postcode validator, it will reject bad postcodes - what i'm doing here is proving it with some trivial examples.

If we're going to use the feature file to literally test-drive out the validation then we need more examples than those offered.

Additionally - I'd talk to the API guys about whether this validation should be done inside forms web app or inside the API... remember: if there's an API you can post data to it, and if you can post data to it you might post invalid data... so protecting the 'client' is less useful than protecting the API.
(That last bit isn't my business, incidentally - so long as the tests are green i'm not fussed  -where- the validation is.. but I suspect the API guys might be..)

Reply:
I changed the regex and added more examples

Reply:
([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})

I added different valid examples throughout the feature file

Reply:
Concerning the point of the API, all the validation feature per feature is not yet done in the backend. We are just dealing with trivial inconsistencies at the moment. Maybe it will change but it will require its own US and it will not be trivial one.